draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -1735,28 +1735,37 @@ deployed and not likely to be registered in a timely manner otherwise.</t>
 <section title="Considerations for New HTTP Fields" anchor="considerations.for.new.header.fields">
 
 <t>
-   See <xref target="field.names"/> for a general requirements for field names,
+   See <xref target="field.names"/> for general requirements of field names,
    and <xref target="field.values"/> for a discussion of field values.
 </t>
 <t>
    Authors of specifications defining new fields are advised to consider
    documenting:
 </t>
 <ul>
-  <li>
-    <t>Whether the field is a single value or whether it can be a list
-    (delimited by commas; see <xref target="field.values"/>).</t>
-    <t>If it is not a list, document how to treat messages
-    where the field occurs multiple times (a sensible default would
-    be to ignore the field, but this might not always be the right
-    choice).</t>
-    <t>Note that intermediaries and software libraries might combine
-    multiple field instances into a single one, despite the
-    field's definition not allowing the list syntax. A robust format enables
-    recipients to discover these situations (good example: "Content-Type",
-    as the comma can only appear inside quoted strings;
-    bad example: "Location", as a comma can occur inside a URI).</t>
-  </li>
+  <li><t>Whether the field is a single value or whether it can be a list
+    (delimited by commas; see <xref target="field.values"/>).</t></li>
+    <ul>
+    <li><t>If it is not a list, document how to treat messages
+      where the field occurs multiple times (a sensible default would
+      be to ignore the field, but this might not always be the right
+      choice).</t></li>
+    <li><t>Note that intermediaries and software libraries often combine
+      multiple field instances into a single field value, regardless of the
+      field's definition. A robust value format that distinguishes
+      comma-separated list values from a singular value for the field will
+      allow recipients to more easily discover an improperly repeated field.
+      In contrast, a single value definition that allows an unquoted comma
+      is difficult to distinguish from multiple field lines that have been
+      combined. Hence, "Content-Type" is a good example because a valid comma
+      would only appear inside a quoted string, whereas "Location" is a
+      bad example because a comma might occur inside a valid URI.</t></li>
+    <li><t>When a field definition does use the list syntax, avoid using a
+      list or unquoted comma within the member values, since that makes it
+      very difficult for a recipient to parse the field unambiguously.
+      A bad example for that is the list of credentials in WWW-Authenticate
+      (<xref target="header.www-authenticate"/>).</t></li>
+    </ul>
```

Comment:
Not convinced. This suggests that any use of commas inside field values is bad, unless when it happens in quoted strings. Do we really want to go there? What about etags and URIs?

WWW-Authenticate is indeed tricky to parse, but that's because it uses COMMA as *delimiter* in two different layers (separating challenges, but also separating auth-params inside challenges).